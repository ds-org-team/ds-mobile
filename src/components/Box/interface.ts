import { ReactNode } from 'react';
import { BoxProps } from '@shopify/restyle';
import { Theme } from '../../themes/institucional';

type CustomBoxProps = Partial<{
  children: ReactNode;
  br: Custom.BorderRadius;
  bw: Custom.BorderWidth;
  bbw: Custom.BorderWidth;
  btw: Custom.BorderWidth;
  blw: Custom.BorderWidth;
  brw: Custom.BorderWidth;
  op: Custom.Opacity;
  sof: Custom.Shadow;
  sr: Custom.Shadow;
  sop: Custom.Shadow;
  h: Custom.HeightComponent;
  sz?: Custom.Size;
  style: ViewStyle;
  testID: string;
}> &
  Partial<BoxProps<Theme>>;

export { CustomBoxProps };
