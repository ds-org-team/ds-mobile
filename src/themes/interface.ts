export type ITokens = {
  colors: {
    transparent: string;
    background: {
      default: string;
      brand: string;
    };
    fittings: {
      text: {
        primary: {
          enabled: string;
          disabled: string;
        };
        secondary: {
          enabled: string;
          disabled: string;
        };
        inverse: {
          enabled: string;
          disabled: string;
        };
      };
      border: {
        primary: {
          enabled: string;
          disabled: string;
        };
        inverse: {
          enabled: string;
          disabled: string;
        };
      };
      icon: {
        primary: {
          enabled: string;
          disabled: string;
        };
        inverse: {
          enabled: string;
          disabled: string;
        };
      };
      divider: string;
    };
    surface: {
      default: string;
      pressed: string;
      select: string;
      disabled: string;
      overlay: string;
    };
    action: {
      main: {
        primary: string;
        pressed: string;
      };
      confirm: {
        primary: string;
        pressed: string;
      };
      remove: {
        primary: string;
        pressed: string;
      };
      help: {
        primary: string;
        pressed: string;
      };
    };
    feedback: {
      positive: {
        fill: string;
        border: string;
        icon: string;
        text: string;
      };
      negative: {
        fill: string;
        border: string;
        icon: string;
        text: string;
      };
      notice: {
        fill: string;
        border: string;
        icon: string;
        text: string;
      };
      informative: {
        fill: string;
        border: string;
        icon: string;
        text: string;
      };
    };
    accent: {
      fill: string;
      border: string;
      icon: string;
      text: string;
    };
    decorative: {
      one: {
        primary: string;
        inverse: string;
      };
      two: {
        primary: string;
        inverse: string;
      };
      tree: {
        primary: string;
        inverse: string;
      };
      four: {
        primary: string;
        inverse: string;
      };
      five: {
        primary: string;
        inverse: string;
      };
    };
  };
  spacing: {
    quark: number;
    nano: number;
    xs: number;
    sm: number;
    md: number;
    lg: number;
  };
  screen: {
    breakpoints: {
      phone: number;
      tablet: number;
    };
  };
  font: {
    variants: {
      bold: {
        fontFamily: string;
        fontWeight: string;
      };
      semibold: {
        fontFamily: string;
        fontWeight: string;
      };
      medium: {
        fontFamily: string;
        fontWeight: string;
      };
      regular: {
        fontFamily: string;
        fontWeight: string;
      };
    };
  };
  radius: {
    none: number;
    nano: number;
    sm: number;
    md: number;
    lg: number;
    pill: number;
    circular: number;
  };
};

export type ITheme = {
  borderRadii: {
    circular: number;
    lg: number;
    md: number;
    nano: number;
    none: number;
    pill: number;
    sm: number;
  };
  breakpoints: {
    phone: number;
    tablet: number;
  };
  colors: {
    'accent-border': string;
    'accent-fill': string;
    'accent-icon': string;
    'accent-text': string;
    'action-confirm-pressed': string;
    'action-confirm-primary': string;
    'action-help-pressed': string;
    'action-help-primary': string;
    'action-main-pressed': string;
    'action-main-primary': string;
    'action-remove-pressed': string;
    'action-remove-primary': string;
    'background-brand': string;
    'background-default': string;
    'decorative-five-inverse': string;
    'decorative-five-primary': string;
    'decorative-four-inverse': string;
    'decorative-four-primary': string;
    'decorative-one-inverse': string;
    'decorative-one-primary': string;
    'decorative-tree-inverse': string;
    'decorative-tree-primary': string;
    'decorative-two-inverse': string;
    'decorative-two-primary': string;
    'feedback-informative-border': string;
    'feedback-informative-fill': string;
    'feedback-informative-icon': string;
    'feedback-informative-text': string;
    'feedback-negative-border': string;
    'feedback-negative-fill': string;
    'feedback-negative-icon': string;
    'feedback-negative-text': string;
    'feedback-notice-border': string;
    'feedback-notice-fill': string;
    'feedback-notice-icon': string;
    'feedback-notice-text': string;
    'feedback-positive-border': string;
    'feedback-positive-fill': string;
    'feedback-positive-icon': string;
    'feedback-positive-text': string;
    'fittings-border-inverse-disabled': string;
    'fittings-border-inverse-enabled': string;
    'fittings-border-primary-disabled': string;
    'fittings-border-primary-enabled': string;
    'fittings-divider': string;
    'fittings-icon-inverse-disabled': string;
    'fittings-icon-inverse-enabled': string;
    'fittings-icon-primary-disabled': string;
    'fittings-icon-primary-enabled': string;
    'fittings-text-inverse-disabled': string;
    'fittings-text-inverse-enabled': string;
    'fittings-text-primary-disabled': string;
    'fittings-text-primary-enabled': string;
    'fittings-text-secondary-disabled': string;
    'fittings-text-secondary-enabled': string;
    'surface-default': string;
    'surface-disabled': string;
    'surface-overlay': string;
    'surface-pressed': string;
    'surface-select': string;
    transparent: string;
  };
  spacing: {
    lg: number;
    md: number;
    nano: number;
    quark: number;
    sm: number;
    xs: number;
  };
  textVariants: {
    bold: {
      fontFamily: string;
      fontWeight: number;
    };
    medium: {
      fontFamily: string;
      fontWeight: number;
    };
    regular: {
      fontFamily: string;
      fontWeight: number;
    };
    semibold: {
      fontFamily: string;
      fontWeight: number;
    };
  };
};
