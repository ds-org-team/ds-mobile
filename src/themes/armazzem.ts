import { extendTheme } from '../utils/theme';
import { ITokens } from './interface';
import themeInstitucional from './institucional';

const tokens: Partial<ITokens> = {
  colors: {
    transparent: 'transparent',
    background: {
      default: '#FFFFFF',
      brand: '#B68500',
    },
    fittings: {
      text: {
        primary: {
          enabled: '#1A233D',
          disabled: '#8F96A8',
        },
        secondary: {
          enabled: '#58627F',
          disabled: '#DADDE5',
        },
        inverse: {
          enabled: '#FFFFFF',
          disabled: '#8F96A8',
        },
      },
      border: {
        primary: {
          enabled: '#58627F',
          disabled: '#DADDE5',
        },
        inverse: {
          enabled: '#FFFFFF',
          disabled: '#8F96A8',
        },
      },
      icon: {
        primary: {
          enabled: '#58627F',
          disabled: '#DADDE5',
        },
        inverse: {
          enabled: '#FFFFFF',
          disabled: '#8F96A8',
        },
      },
      divider: '#BDBDBD',
    },
    surface: {
      default: '#FFFFFF',
      pressed: '#D1E3FE',
      select: '#E7F0FE',
      disabled: '#F0F2FA',
      overlay: '#58627F',
    },
    action: {
      main: {
        primary: '#B68500',
        pressed: '#795300',
      },
      confirm: {
        primary: '#15AB51',
        pressed: '#066347',
      },
      remove: {
        primary: '#D63429',
        pressed: '#7C0D21',
      },
      help: {
        primary: '#5487F9',
        pressed: '#0A7E19',
      },
    },
    feedback: {
      positive: {
        fill: '#CEFACE',
        border: '#15AB51',
        icon: '#15AB51',
        text: '#15AB51',
      },
      negative: {
        fill: '#FCE4D4',
        border: '#D63429',
        icon: '#D63429',
        text: '#D63429',
      },
      notice: {
        fill: '#FFEECE',
        border: '#FF780E',
        icon: '#FF780E',
        text: '#FF780E',
      },
      informative: {
        fill: '#0A7E19',
        border: '#0A7E19',
        icon: '#0A7E19',
        text: '#0A7E19',
      },
    },
    accent: {
      fill: '#0A7E19',
      border: '#0A7E19',
      icon: '#0A7E19',
      text: '#0A7E19',
    },
    decorative: {
      one: {
        primary: '#FEF8CB',
        inverse: '#795300',
      },
      two: {
        primary: '#FFE3EF',
        inverse: '#65167A',
      },
      tree: {
        primary: '#F9E0FF',
        inverse: '#38137A',
      },
      four: {
        primary: '#E1FDF9',
        inverse: '#13436D',
      },
      five: {
        primary: '#F4FCCF',
        inverse: '#396504',
      },
    },
  },
};

const theme = extendTheme(tokens, themeInstitucional);

export default theme;
