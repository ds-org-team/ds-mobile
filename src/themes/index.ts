export { default as themeInstitucional } from './institucional';
export * from './institucional';
export * from './properties';
export { ThemeProvider, useTheme } from '@shopify/restyle';
export { ITheme, ITokens } from './interface';
export { createTheme, extendTheme } from '../utils/theme';
