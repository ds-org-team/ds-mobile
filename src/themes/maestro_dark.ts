import { createTheme } from '../utils/theme';

const tokens = {
  colors: {
    background: {
      default: '#0C1224',
      brand: '#1D5AF5',
    },
    fittings: {
      text: {
        primary: {
          enabled: '#DADDE5',
          disabled: '#58627F',
        },
      },
    },
    surface: {
      default: '#FFFFFF',
      pressed: '#D1E3FE',
      select: '#E7F0FE',
    },
    action: {
      primary: '#5487F9',
      'primary-pressed': '#5487F9',
      confirm: '#15AB51',
      'confirm-pressed': '#15AB51',
      remove: '#D63429',
      'remove-pressed': '#D63429',
      help: '#1D5AF5',
      'help-pressed': '#1D5AF5',
    },
    feedback: {
      positive: {
        fill: '#CEFACE',
        border: '#15AB51',
        icon: '#15AB51',
        text: '#15AB51',
      },
      negative: {
        fill: '#FCE4D4',
        border: '#D63429',
        icon: '#D63429',
        text: '#D63429',
      },
      notice: {
        fill: '#FFEECE',
        border: '#FF780E',
        icon: '#FF780E',
        text: '#FF780E',
      },
      informative: {
        fill: '#D1E3FE',
        border: '#1D5AF5',
        icon: '#1D5AF5',
        text: '#1D5AF5',
      },
    },
    accent: {
      fill: '#1D5AF5',
      border: '#1D5AF5',
      icon: '#1D5AF5',
      text: '#1D5AF5',
    },
    black: '#000',
    white: '#fff',
    neutral: {
      lightest: '#fff',
      dark: '#212121',
    },
  },
  spacing: {
    quark: 4,
    nano: 8,
    xs: 16,
    sm: 24,
    md: 32,
    lg: 40,
  },
  sizing: {
    xs: 16,
    sm: 20,
    md: 24,
    lg: 32,
    xl: 48,
  },
  screen: {
    breakpoints: {
      phone: 0,
      tablet: 768,
    },
  },
  font: {
    variants: {
      bold: {
        fontFamily: '"Public Sans", sans-serif',
        fontWeight: '700',
      },
      semibold: {
        fontFamily: '"Public Sans", sans-serif',
        fontWeight: '600',
      },
      medium: {
        fontFamily: '"Public Sans", sans-serif',
        fontWeight: '500',
      },
      regular: {
        fontFamily: '"Public Sans", sans-serif',
        fontWeight: '400',
      },
    },
    'line-height': {
      lg: 32,
      md: 25.6,
      sm: 19.2,
      xs: 16,
    },
    'letter-spacing': {
      lg: 1.5,
      md: 0.5,
      sm: 0,
      xs: -1.5,
    },
    size: {
      xs: 16,
      sm: 20,
      md: 24,
      lg: 32,
      xl: 48,
    },
  },
  radius: {
    none: 0,
    nano: 4,
    sm: 8,
    md: 16,
    lg: 24,
    pill: 500,
    circular: 9999,
  },
  opacity: {
    'opacity-80': 0.8,
    'opacity-64': 0.64,
    'opacity-32': 0.32,
    'opacity-16': 0.16,
    'opacity-08': 0.08,
  },
  shadows: {
    offset: {
      xs: { width: 0, height: 1.5 },
      sm: { width: 0, height: 4 },
      md: { width: 0, height: 10 },
      lg: { width: 0, height: 16 },
    },
    radius: {
      xs: 3.5,
      sm: 8,
      md: 16,
      lg: 32,
    },
    opacity: {
      xs: 0.12,
      sm: 0.1,
      md: 0.1,
      lg: 0.1,
    },
  },
  border: {
    width: {
      none: 0,
      xs: 1,
      sm: 2,
      md: 4,
      lg: 8,
    },
    radius: {
      none: 0,
      nano: 4,
      sm: 8,
      md: 16,
      lg: 24,
      pill: 500,
      circular: 9999,
    },
  },
};

const theme = createTheme(tokens);

export default theme;
