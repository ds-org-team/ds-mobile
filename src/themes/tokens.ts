export const BORDER_WIDTH = ['none', 'xs', 'sm', 'md', 'lg'] as const;

export const RADIUS = [
  'none',
  'nano',
  'sm',
  'md',
  'lg',
  'pill',
  'circular',
] as const;

export const COLORS = [
  'accent-border',
  'accent-fill',
  'accent-icon',
  'accent-text',
  'action-confirm',
  'action-confirm-pressed',
  'action-help',
  'action-help-pressed',
  'action-primary',
  'action-primary-pressed',
  'action-remove',
  'action-remove-pressed',
  'background-brand',
  'background-default',
  'black',
  'feedback-informative-border',
  'feedback-informative-fill',
  'feedback-informative-icon',
  'feedback-informative-text',
  'feedback-negative-border',
  'feedback-negative-fill',
  'feedback-negative-icon',
  'feedback-negative-text',
  'feedback-notice-border',
  'feedback-notice-fill',
  'feedback-notice-icon',
  'feedback-notice-text',
  'feedback-positive-border',
  'feedback-positive-fill',
  'feedback-positive-icon',
  'feedback-positive-text',
  'fittings-text-primary-disabled',
  'fittings-text-primary-enabled',
  'neutral-dark',
  'neutral-lightest',
  'surface-default',
  'surface-pressed',
  'surface-select',
  'white',
];

export const FONT_SIZES = [
  'xs',
  'sm',
  'md',
  'lg',
  'xl',
  '2xl',
  '3xl',
  '4xl',
  '5xl',
  '6xl',
  '7xl',
  '8xl',
  '9xl',
  '10xl',
] as const;

export const HEIGHTS = ['xs', 'sm', 'md', 'lg', 'xl'] as const;

export const LINE_HEIGHTS = ['lg', 'md', 'sm', 'xs'] as const;

export const LETTER_SPACINGS = ['lg', 'md', 'sm', 'xs'] as const;

export const OPACITIES = [
  'opacity-5',
  'opacity-4',
  'opacity-3',
  'opacity-2',
  'opacity-1',
] as const;

export const SHADOWS = ['xs', 'sm', 'md', 'lg'] as const;

export const SPACINGS = [
  'xs',
  'sm',
  'md',
  'lg',
  'xl',
  '2xl',
  '3xl',
  '4xl',
  '5xl',
  '6xl',
  '7xl',
  '8xl',
  '9xl',
] as const;

export const SIZES = ['xs', 'sm', 'md', 'lg', 'xl', '2xl'] as const;
