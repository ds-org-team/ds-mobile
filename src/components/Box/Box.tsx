import { createBox, createRestyleComponent } from '@shopify/restyle';
import React from 'react';
import {
  borderRadius,
  borderWidth,
  borderBottomWidth,
  borderTopWidth,
  borderLeftWidth,
  borderRightWidth,
  opacity,
  shadowOffset,
  shadowOpacity,
  shadowRadius,
  heightComponent,
  widthComponent,
} from '../../themes';
import { ITheme } from '../../themes/interface';
import { CustomBoxProps } from './interface';

const Box = createBox<ITheme>();

const CustomBox: React.FC<CustomBoxProps> = ({
  children,
  testID,
  ...props
}) => (
  <Box testID={testID || 'Box'} {...props}>
    {children}
  </Box>
);

export default createRestyleComponent<CustomBoxProps, ITheme>(
  [
    borderRadius,
    borderWidth,
    borderBottomWidth,
    borderTopWidth,
    borderLeftWidth,
    borderRightWidth,
    opacity,
    shadowOffset,
    shadowRadius,
    shadowOpacity,
    heightComponent,
    widthComponent,
  ],
  CustomBox,
);
