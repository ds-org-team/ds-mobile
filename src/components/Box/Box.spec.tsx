import React from 'react';
import { cleanup, render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import { themeInstitucional } from '../../themes';

import Box from './Box';

afterEach(cleanup);

test('should be action-main-primary as background-color', async () => {
  const { toJSON } = render(
    <ThemeProvider theme={themeInstitucional}>
      <Box backgroundColor="action-main-primary" />
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});

test('should be large as margin', async () => {
  const { toJSON } = render(
    <ThemeProvider theme={themeInstitucional}>
      <Box margin="lg" />
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});

test('should be Box render correctly', async () => {
  const { toJSON } = render(<Box />);

  expect(toJSON()).toMatchSnapshot();
});
