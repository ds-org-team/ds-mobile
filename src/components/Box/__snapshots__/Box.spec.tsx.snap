// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be Box render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
  testID="Box"
/>
`;

exports[`should be action-main-primary as background-color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1D5AF5",
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
  testID="Box"
/>
`;

exports[`should be large as margin 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "margin": 40,
      },
      Array [
        Object {},
      ],
    ]
  }
  testID="Box"
/>
`;
